frontend/src/components/import-button.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { Button, Dropdown, DropdownButton, Modal } from "react-bootstrap";
+import { readFile } from "../libs/fileManager";
+
+export class ImportButton extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: null,
+            dialogContents: "",
+            dialogOpen: false
+        };
+    }
+
+    /* assume frontend data is up to date
+     *  upload the assignment object
+     *  if there's an part of inconsistency between the imported data and frontend data
+     *  then apiGET that part of data and re-verify it
+     */
+
+    handleClose = () => {
+        this.setState({ dialogOpen: false });
+    };
+
+    uploadDataToBackend = () => {
+        // TODO
+    };
+
+    importFile = () => {
+        let loadDataFunc = data => {
+            // passed in data is of json format
+            console.log(data);
+
```

Comment:
Again, print or throw an error. This function should probably be `importClicked` or some such. I doubt any import logic should happen here.